app/controllers/api/pxe_servers_controller.rb

```diff
@@ -1,5 +1,91 @@
 module Api
   class PxeServersController < BaseController
+    INVALID_PXE_SERVER_ATTRS = %w[id href].freeze # Cannot update or create these
+    REQUIRED_PXE_SERVER_ATTRS = %w[name uri].freeze
+    REQUIRED_PXE_MENU_ATTRS = %w[file_name].freeze
+    REQUIRED_PXE_AUTH_ATTRS = %w[userid password].freeze
+
     include Subcollections::PxeImages
+    include Subcollections::PxeMenus
+
+    def create_resource(_type, _id, data = {})
+      PxeServer.transaction do
+        authentication = data.delete('authentication')
```

Comment:
is it possible to put code outside of `PxeServer.transaction` block which is not related to the transaction ? (same for edit)